src/interface/IService.ts

```diff
@@ -10,4 +10,6 @@ export default interface IService extends IDestructable {
      * Implement the final processing of logs before sending it directly to the endpoint
      */
     preparePayload<T>(logs: T[]): Promise<any>;
+
+    serializer<T>(log: T): any;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type. ([typescript:S4204](https://sonarcloud.io/organizations/alexeypopovua-github/rules#rule_key=typescript%3AS4204))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=advanced-logger&pullRequest=75&issues=AWfltDh5rt4Kct_4x7pr&open=AWfltDh5rt4Kct_4x7pr)

[ISSUE_KEY]: AWfltDh5rt4Kct_4x7pr